virtual_ecosystem/models/animal/animal_cohorts.py

```diff
@@ -239,41 +313,49 @@ def die_individual(self, number_dead: int, carcass_pool: CarcassPool) -> None:
         # Find total mass contained in the carcasses
         carcass_mass = number_dead * self.mass_current
 
-        # Split this mass between carcass decay, and scavengeable carcasses
-        carcass_pool.scavengeable_energy += (
-            1 - self.decay_fraction_carcasses
-        ) * carcass_mass
-        carcass_pool.decomposed_energy += self.decay_fraction_carcasses * carcass_mass
+        self.update_carcass_pool(carcass_mass, carcass_pools)
 
-    def update_carcass_pool(self, carcass_mass: float, carcass_pool: DecayPool) -> None:
-        """Updates the carcass pool based on consumed mass and predator's efficiency.
+    def update_carcass_pool(
+        self, carcass_mass: float, carcass_pools: list[CarcassPool]
+    ) -> None:
+        """Updates the carcass pools based on consumed mass and predator's efficiency.
+
+        TODO: move to animal model?
 
         Args:
             carcass_mass: The total mass consumed from the prey cohort.
-            carcass_pool: The pool to which remains of eaten individuals are delivered.
+            carcass_pools: The pools to which remains of eaten individuals are
+            delivered.
         """
-
-        # Update the carcass pool with the remainder
-        carcass_pool.scavengeable_energy += (
-            1 - self.decay_fraction_carcasses
-        ) * carcass_mass
-        carcass_pool.decomposed_energy += self.decay_fraction_carcasses * carcass_mass
+        number_carcass_pools = len(carcass_pools)
+        carcass_mass_per_pool = carcass_mass / number_carcass_pools
+
+        for carcass_pool in carcass_pools:
```

Comment:
And the same here. Also, why you use a list here for these pools but a set for the `excrement_pools` above?